simulator/src/main/java/com/github/benmanes/caffeine/cache/simulator/policy/linked/LinkedPolicy.java

```diff
@@ -89,22 +108,25 @@ public void record(long key) {
 
   /** Evicts while the map exceeds the maximum capacity. */
   private void evict(Node candidate) {
-    if (data.size() > maximumSize) {
-      Node victim = policy.findVictim(sentinel, policyStats);
-      policyStats.recordEviction();
-
-      boolean admit = admittor.admit(candidate.key, victim.key);
-      if (admit) {
-        evictEntry(victim);
-      } else {
-        evictEntry(candidate);
+    if (currentSize > maximumSize) {
+      while (currentSize > maximumSize) {
```

Comment:
If the candidate is evicted, but further evictions are needed, then you may re-evict the candidate. When doing so you decrease the current size multiple times, but only removed once. This means that the state is corrupted and no longer bounded as expected.

I believe in your case you would need to evict only the victims once the candidate was discarded.

Reply:
When the candidate is evicted we guaranteed that no more evictions needed, since the candidate weight is the reason we exceed the maximumSize in the first place. Aren't we?

Reply:
you're right, I was thinking about W-TinyLFU where this is delayed a bit.